api/v1beta1/machinedeployment_types.go

```diff
@@ -25,6 +25,10 @@ const (
 	// MachineDeploymentTopologyFinalizer is the finalizer used by the topology MachineDeployment controller to
```

Comment:
One quick general comment (might take a few days until I get around to a full review).

I think it would be great if we could introduce a new e2e test with this PR. A few notes:
* Scope/name of the test would be ClusterDelete
* Tests steps (for now):
  * Create Cluster
  * Add finalizers on MD / MS / Machines  / InfraCluster / ControlPlane (we can open an issue to follow-up for MPs)
  * Trigger delete Cluster
  * Ensure step-by-step that:
    * Only the expected objects have a deletionTimestamp
    * Then unblock these by removing the finalizer
    * Repeat

In later PRs and once the corresponding features are implemented we can extend the test coverage for:
* Delete conditions
* Propagation of timeouts on deleting objects

WDYT?

Happy to brainstorm about this a bit if necessary

Reply:
+1 
considering that the proposed test introduce coverage for the entire deletion workflow, not only for MD I'm  ok also in implementing it as a follow up, but having it now will really help in validating this PR

Reply:
Definitely fine with a follow-up in general.

Would be good to at least have the verification of the MD / MS / Machine deletion flow already in this PR so we have coverage of the logic introduced in this PR